### ISSUE TITLE
S3: Multi-part upload fails signature check

### ISSUE THREAD

**Issue by ochrons** [aws/aws-sdk-java-v2#958]
## Expected Behavior
Creating a new multi-part upload should work ok.

## Current Behavior
Creating a new multi-part upload fails signature check
```
software.amazon.awssdk.services.s3.model.S3Exception: The request signature we calculated does not match the signature you provided. Check your key and signing method.
```

## Possible Solution
It seems that `uploads` query parameter is missing from the HTTP request. This might explain why the signature does not match.
```
Sending Request: DefaultSdkHttpFullRequest(httpMethod=POST, protocol=https, host=bucketname.s3.us-east-2.amazonaws.com, port=443, encodedPath=/2d12b1d999734ab9fcbeba621ccc3efce62c5ef4, headers=[amz-sdk-invocation-id, Cache-Control, Content-Length, Content-MD5, Content-Type, User-Agent], queryParameters=[])
```

## Context
Regular get and put object requests work just fine in the same setup, only the multi-part uploads fail.

## Your Environment

* AWS Java SDK version used: 2.1.4
* JDK version used: JDK8
* Operating System and version:  Win10

**Comment by zoewangg** [comment]
I was not able to reproduce the issue. Could you provide the sample code?

**Comment by ochrons** [comment]
Here's a simplified version that fails with invalid signature:

```scala
import software.amazon.awssdk.services.s3.S3AsyncClient
import software.amazon.awssdk.services.s3.model._

import scala.compat.java8.FutureConverters._
import scala.concurrent.Await
import scala.concurrent.duration.Duration

val bucketName  = "your-bucket-here"
val path        = "123456test"
val contentType = "application/octet-stream"

val s3Client = S3AsyncClient.builder().build()

try {
  val req = CreateMultipartUploadRequest
    .builder()
    .bucket(bucketName)
    .key(path)
    .contentType(contentType)
    .cacheControl(s"max-age=3600,s-maxage=3600")

  val result   = Await.result(s3Client.createMultipartUpload(req.build).toScala, Duration.Inf)
  val uploadId = result.uploadId()

  val req2 = AbortMultipartUploadRequest
    .builder()
    .bucket(bucketName)
    .key(path)
    .uploadId(uploadId)
    .build

  val abortResult = Await.result(s3Client.abortMultipartUpload(req2).toScala, Duration.Inf)
} catch {
  case e: Throwable => println(e)
}
```

Log:
```
11:39:59.850 DEBUG [software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain:84] Creating an interceptor chain that will apply interceptors in the following order: [software.amazon.awssdk.services.s3.internal.handlers.EndpointAddressInterceptor@2c5711ad, software.amazon.awssdk.services.s3.internal.handlers.CreateBucketInterceptor@1848aa31, software.amazon.awssdk.services.s3.internal.handlers.PutObjectInterceptor@4948a8b7, software.amazon.awssdk.services.s3.internal.handlers.CreateMultipartUploadRequestInterceptor@79f9a15b, software.amazon.awssdk.services.s3.internal.handlers.EnableChunkedEncodingInterceptor@6db70db0, software.amazon.awssdk.services.s3.internal.handlers.DisableDoubleUrlEncodingInterceptor@30f32b15, software.amazon.awssdk.services.s3.internal.handlers.DecodeUrlEncodedResponseInterceptor@3c7220ac, software.amazon.awssdk.services.s3.internal.handlers.AddContentMd5HeaderInterceptor@835ad00, software.amazon.awssdk.services.s3.internal.handlers.GetBucketPolicyInterceptor@5dbaeb3b, software.amazon.awssdk.services.s3.internal.handlers.AsyncChecksumValidationInterceptor@7438d81a, software.amazon.awssdk.services.s3.internal.handlers.SyncChecksumValidationInterceptor@59e2f171, software.amazon.awssdk.services.s3.internal.handlers.EnableTrailingChecksumInterceptor@7c9a0ae0, software.amazon.awssdk.services.s3.internal.handlers.ExceptionTranslationInterceptor@7673d57b]
11:39:59.852 DEBUG [software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain:84] Interceptor 'software.amazon.awssdk.services.s3.internal.handlers.EndpointAddressInterceptor@2c5711ad' modified the message with its modifyHttpRequest method.
11:39:59.852 DEBUG [software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain:84] Interceptor 'software.amazon.awssdk.services.s3.internal.handlers.CreateMultipartUploadRequestInterceptor@79f9a15b' modified the message with its modifyHttpRequest method.
11:39:59.853 DEBUG [software.amazon.awssdk.core.interceptor.ExecutionInterceptorChain:84] Interceptor 'software.amazon.awssdk.services.s3.internal.handlers.AddContentMd5HeaderInterceptor@835ad00' modified the message with its modifyHttpRequest method.
11:39:59.854 DEBUG [software.amazon.awssdk.request:84] Sending Request: DefaultSdkHttpFullRequest(httpMethod=POST, protocol=https, host=bucket-name-elided.s3.us-east-2.amazonaws.com, port=443, encodedPath=/123456test, headers=[amz-sdk-invocation-id, Content-Length, Content-MD5, Content-Type, User-Agent], queryParameters=[])
11:39:59.855 DEBUG [software.amazon.awssdk.auth.signer.Aws4Signer:84] AWS4 String to sign: AWS4-HMAC-SHA256
20190102T094001Z
20190102/us-east-2/s3/aws4_request
c68e6b46940f6791064e343c5c18d130db2744010d32a1aeea1b13f65514a078
java.util.concurrent.CompletionException: software.amazon.awssdk.services.s3.model.S3Exception: The request signature we calculated does not match the signature you provided. Check your key and signing method. (Service: S3, Status Code: 403, Request ID: 99CA2CB97F48182D)
res0: Unit = ()
```

**Comment by zoewangg** [comment]
Thanks for providing the sample code. I was able to reproduce the issue. It appears `S3Client` is working fine and only`S3AsyncClient` has this issue.

**Comment by zoewangg** [comment]
Fix has been released in `2.3.2`
